taker/src/routes.rs

```diff
@@ -267,12 +271,12 @@ pub struct WithdrawRequest {
 }
 
 #[rocket::post("/withdraw", data = "<withdraw_request>")]
-#[instrument(name = "POST /withdraw", skip(taker, _auth), err)]
+#[instrument(name = "POST /withdraw", skip(taker, _user), err)]
 pub async fn post_withdraw_request(
     withdraw_request: Json<WithdrawRequest>,
     taker: &State<Taker>,
     network: &State<Network>,
-    _auth: Authenticated,
+    _user: User,
```

Comment:
🔧 I think you can remove `_user: User` from all these routes where it isn't used.

Reply:
No, it's needed because this is the guard. The field `user` will only be available if the user is logged in.

Reply:
Oh, interesting. That's not obvious at all. How does that work?

Reply:
To be precise, it's a [`RequestGuard`](https://rocket.rs/v0.4/guide/requests/#request-guards). We implement `FromRequest` for this type here: https://github.com/itchysats/itchysats/blob/82b32f1d6388881bd471243411ee8a3810682070/rocket-cookie-auth/src/user.rs#L23 

It's similar to before with `Authenticate` which implemented `FromRequest` here: https://github.com/itchysats/itchysats/blob/ea3170f2da4fcbbdf9b6152d84cd1ab4bcc3b6d6/rocket-basicauth/src/lib.rs#L81